rpc/src/v1/helpers/errors.rs

```diff
@@ -444,7 +445,41 @@ pub fn filter_block_not_found(id: BlockId) -> Error {
 	}
 }
 
+pub fn on_demand_error(err: OnDemandError) -> Error {
+	match err.kind() {
+		OnDemandErrorKind::ChannelCanceled(ref e) => return on_demand_cancel(e.clone()),
```

Comment:
Needless `return`

Also, is it possible to take ownership of the `future::oneshot::Canceled´ instead and not clone it?

Reply:
Should be possible with a `match err { Error(OnDemandErrorKind::ChannelCanceled(e), _) =>  e`